### ISSUE TITLE
Tapping into animation cycle

### ISSUE THREAD

**Issue by fernandocanizo** [VincentGarreau/particles.js#34]
Hello Vincent, I was trying to use your library like it can be seen here: http://flc.muriandre.com/ which is changing shapes and colors after several drawing rounds in order to show more effects by one single script.

That example is using your library with my modifications that you can get from here: https://github.com/fernandocanizo/particles.js

My first attempt was to just create another animation loop, problem was machine became sluggish as each new frame I created leaved your requested animations running (but not being shown), so at some point there were just too many animations running for the machine to cope with.

So I decided to export the **id** returned by **requestAnimFrame()** and clear the previous call when needed.

Then I needed a way to change configuration parameters and I tried several things until I decided to export your config object too.

Then I needed a way to tap into your animation cycle, so I exported a couple of variables that allow to pass it a callback which I use to update config.

Then I needed to recalculate colors, since they are only calculated on particle creation.

...

Long story short: I did a lot of changes to your library structure, but I think they would be good to be included in your library, but now I've gone to far and it seems you've been updating it too, so I'm wondering: supposing you like my changes, how would you like we to proceed to incorporate them? Should I just make a pull request? Should I update first and then try to trace your changes and include them in my version?

Please tell me what you think of my changes and if you like them, how we can proceed.
